### PR TITLE
Correct the `get_mut` and `get_mut_or_default` for `ByteMapView`

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

For those two functions, the `deleted_prefixes` were not handled which is a mistake.

## Proposal

While being at that some other things were done:
* The `.unwrap()` were replaced by `?` and the use of `Result<()>` or similar.
* The random tests were grouped in the same file which allow to take advantage of the parallelism of the tests and a little faster CI.
* The API of `get_mut` and similar functions in `CollectionView`/ `ReentrantCollectionView` was changed so that it now takes a `&[u8]` instead of `Vec<u8>` for simplicity.
* The helper function `load_value` was eliminated.

## Test Plan

A specific test that repro

## Release Plan

The internal documentation of the functions was changed to reflect the new API.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-views/src/map_view.rs [hunk 1]
```diff
@@ -713,22 +710,26 @@ where
...
-    ///   let value = map.get_mut_or_default(vec![0,1]).await.unwrap();
+    ///   assert_eq!(map.get_mut_or_default(&[7]).await.unwrap(), "");
+    ///   let value = map.get_mut_or_default(&[0,1]).await.unwrap();
     ///   assert_eq!(*value, String::from("Hello"));
     ///   *value = String::from("Hola");
     ///   assert_eq!(map.get(&[0,1]).await.unwrap(), Some(String::from("Hola")));
     /// # })
     /// ```
-    pub async fn get_mut_or_default(&mut self, short_key: Vec<u8>) -> Result<&mut V, ViewError> {
+    pub async fn get_mut_or_default(&mut self, short_key: &[u8]) -> Result<&mut V, ViewError> {
```
Comment (ma2bd) [inline linera-views/src/map_view.rs:720]: nit: `get_mut` and `get_mut_or_default` are written differently which is a little weird. Apparently:
* `get_mut` is ok accessing `self.updates` several times
* `get_mut_or_default` tries not to do that

File: linera-views/src/map_view.rs [hunk 2]
```diff
@@ -713,22 +710,26 @@ where
...
         use std::collections::btree_map::Entry;
 
-        let update = match self.updates.entry(short_key.clone()) {
+        let update = match self.updates.entry(short_key.to_vec()) {
             Entry::Vacant(e) if self.delete_storage_first => e.insert(Update::Set(V::default())),
             Entry::Vacant(e) => {
-                let key = self.context.base_index(&short_key);
-                let value = self.context.read_value(&key).await?.unwrap_or_default();
-                e.insert(Update::Set(value))
+                if contains_key(&self.deleted_prefixes, short_key) {
```
Comment (ma2bd) [inline linera-views/src/map_view.rs:726]: This case could be shared above `if self.delete_storage_first || contains_key(&self.deleted_prefixes, short_key) `

**Review by ma2bd — COMMENTED** [review]
Thanks for the fix. I'm assuming that only `get_mut` and `get_mut_or_default` are changed meaningfully

File: linera-views/src/map_view.rs (hunk 1 above)
Comment (MathieuDutSik) [reply linera-views/src/map_view.rs:720]: It was possible indeed to correct that.

File: linera-views/src/map_view.rs (hunk 2 above)
Comment (MathieuDutSik) [reply linera-views/src/map_view.rs]: Yes, that is correct,

File: linera-views/src/map_view.rs [hunk 3]
```diff
@@ -319,22 +307,32 @@ where
...
+                    let key = self.context.base_index(short_key);
+                    let value = self.context.read_value(&key).await?;
+                    value.map(|value| e.insert(Update::Set(value)))
+                }
+            }
+            Entry::Occupied(e) => Some(e.into_mut()),
+        };
+        Ok(match update {
+            None => None,
+            Some(update) => match update {
```
Comment (ma2bd) [inline linera-views/src/map_view.rs:331]: nit: `Some(Update::Removed) => None` etc to avoid nested matches (the compiler will re-optimize)

**Review by ma2bd — APPROVED** [review]

**Pull request merged**